TomTom SDK Examples/Examples/UseCase/Offline/Utils/OfflineMapHelper.swift

```diff
@@ -0,0 +1,137 @@
+
+enum OfflineMapHelper {
+    /// Copy the 'Map' folder (all folders and files recursively inside) from the Bundle to the device
+    static func copyFolders(completion: (_ success: Bool, _ error: Error?) -> Void) {
+        let fileManager = FileManager.default
+        
+        guard let pathFromBundle = Bundle.main.resourceURL?.appendingPathComponent(OfflineConfig.mapFolderName).path else {
+            completion(false, OfflineMapError.bundleResourceURLNotExist)
+            return
+        }
+        
+        guard let docsURL = fileManager.urls(for: .documentDirectory, in: .userDomainMask).first else {
+            completion(false, OfflineMapError.documentsDirectoryNotExist)
+            return
+        }
+        let pathDestDocs = docsURL.appendingPathComponent(OfflineConfig.mapFolderName).path
+
+        if fileManager.fileExists(atPath: pathDestDocs) {
+            /// It has already been copied previously
+            /* YOUR CODE GOES HERE */
+            completion(true, nil)
+        } else {
+            /// Copy directory content recursively
+            do {
+                let files = try fileManager.contentsOfDirectory(atPath: pathFromBundle)
+                try? fileManager.copyItem(atPath: pathFromBundle, toPath: pathDestDocs)
+
+                for fileName in files {
+                    try? fileManager.copyItem(atPath: "\(pathFromBundle)/\(fileName)", toPath: "\(pathDestDocs)/\(fileName)")
+                }
+                
+                if OfflineMapPath.mapDataPath == nil {
+                    completion(false, OfflineMapError.mapFolderDoesNotExist)
+                    return
+                }
+                
+                if OfflineMapPath.keystorePath == nil {
+                    completion(false, OfflineMapError.keystoreFileDoesNotExist)
+                    return
+                }
+                
+                completion(true, nil)
+            } catch {
+                completion(false, error)
+            }
+        }
+    }
+}
+
+// MARK: Offline Map Paths
+
+enum OfflineMapPath {
+    // MARK: Internal
+
+    // Properties to access Onboard Map resources
+    static let mapDataPath: String? = getMapDataPath()
+    static let keystorePath: String? = getKeystorePath()
+    static let updateStoragePath: String? = getUpdateStoragePath()
+    static let persistantStoragePath: String? = getPersistantStoragePath()
+
+    // MARK: Private
+
+    private static let resourcesPath: String = Bundle.main.bundlePath
+
+    private static func getUpdateStoragePath() -> String? {
+        let suffix = OfflineConfig.updateStoragePath
+        if let documentDirectory = FileManager.default.urls(for: .documentDirectory, in: .userDomainMask).first {
+            let updateStoragePath = documentDirectory.path + suffix
+            return updateStoragePath
+        } else {
+            return nil
+        }
+    }
+
+    private static func getPersistantStoragePath() -> String? {
```

Comment:
Are we really fetching anything? (e.g. side effect that we want to show by adding get). If not I'd recommend to use a computed variable without `get. Get isn't that common name in Swift it's usually better to use specific word to emphasize size effect (load, request) or remove `get` if there is no side effect.

Reply:
Thank you for providing such a valuable review. I have refactored the codes based on your feedback.